src/common/worker-wrapper.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * This interface establishes a communication protocal between the UI thread and the worker thread
+ */
+export interface WorkerWrapper {
+  /**
+   * Save a file into IndexedDB
+   */
+  saveFile(name: string, file: File): Promise<ArrayBuffer>;
+  /**
+   * Compile the lexical model using files in the IndexedDB
+   */
+  compileModel(): Promise<string>;
+}
```

Comment:
I REALLY like exposing the worker's interface this way! 👍 It's very clean!

Though, now I think we're bumping up against [the second most difficult problem in computer science](https://martinfowler.com/bliki/TwoHardThings.html). 

How about `PredictiveTextStudioWorker`?

```suggestion
export interface PredictiveTextStudioWorker {
  /**
   * Save a file into IndexedDB
   */
  saveFile(name: string, file: File): Promise<ArrayBuffer>;
  /**
   * Compile the lexical model using files in the IndexedDB
   */
  compileModel(): Promise<string>;
}
```

That way the app code uses the `Remote<PredictiveTextStudioWorker>`

Reply:
I have... issues with our current interface... but I think these issues are beyond the scope of this PR 😅